src/decimal/mod.rs

```diff
@@ -346,6 +347,54 @@ dec_impl!(impl_trait_proxy;
         (to_f64; rself;; Option<f64>)
 );
 
+impl<T, P> Sum for GenericDecimal<T, P>
+where
+    T: Clone + GenericInteger + PartialEq,
+    P: Copy + GenericInteger + Into<usize>,
+{
+    fn sum<I: Iterator<Item=Self>>(iter: I) -> Self {
+        iter.fold(GenericDecimal::<T, P>::zero(), Add::add)
+    }
+}
+impl<'a, T, P> Sum<&'a GenericDecimal<T, P>> for GenericDecimal<T, P> where
+    T: Clone + GenericInteger + PartialEq,
+    P: Copy + GenericInteger + Into<usize>,
+{
+    fn sum<I: Iterator<Item=&'a Self>>(iter: I) -> Self {
```

Comment:
I had to implement it this way because otherwise I'd get a really long compilation error that was way too long to debug.